### DISCUSSION TITLE
Mongodb connection string – compressors

### DISCUSSION THREAD

**Discussion by MikePodgorniy** [prisma/prisma#25750] [Q&A] [answered]
### Question

I am trying to use compression settings for MongoDB in my connection string:
mongodb://localhost:27018/db?replicaSet=rs0&compressors=zlib,snappy

However, when I run Prisma, I encounter the following error:

> he provided database string is invalid. MongoDB connection string error: compressors is an invalid option. An option with a similar name exists: compressors in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.

It seems Prisma doesn't support the compressors option. Is there a way to enable compression for MongoDB when using Prisma? If not, are there alternative approaches to achieve this?
 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @MikePodgorniy 👋 

Prisma currently does not support the compression option for MongoDB.  You might want to try setting the compression options directly in your MongoDB server configuration instead of in the connection string. This way, compression would be applied at the server level for all connections, including those made through Prisma.

We have an open [issue](https://github.com/prisma/prisma/issues/20751) to support connection options for MongoDB. Please add your usecase and leave a 👍 to help our engineers prioritize it.

**Comment by MikePodgorniy** [comment]
@ludralph **mongo-rust-driver** https://github.com/prisma/mongo-rust-driver based on official mongodb-rust-driver with full params support, just unlock it or say how to do it, please

**Comment by MikePodgorniy** [comment]
Done, prisma 6.5.0
https://github.com/prisma/prisma-engines/pull/5225

**Reply by nurul3101** [reply]
Thank you for submitting the PR 😄
